zkevm-circuits/src/evm_circuit/execution/callop.rs

```diff
@@ -232,45 +176,37 @@ impl<F: Field> ExecutionGadget<F> for CallOpGadget<F> {
         });
 
         let callee_exists = cb.query_bool();
-        let callee_code_hash = cb.query_cell();
         cb.condition(callee_exists.expr(), |cb| {
             cb.account_read(
-                code_address.expr(),
+                call_gadget.callee_address_expr(),
                 AccountFieldTag::CodeHash,
-                callee_code_hash.expr(),
+                call_gadget.callee_code_hash.expr(),
             );
         });
         cb.condition(1.expr() - callee_exists.expr(), |cb| {
-            cb.account_read(code_address, AccountFieldTag::NonExisting, 0.expr());
+            cb.account_read(
+                call_gadget.callee_address_expr(),
+                AccountFieldTag::NonExisting,
+                0.expr(),
+            );
         });
 
-        let is_empty_code_hash = IsEqualGadget::construct(
-            cb,
-            callee_code_hash.expr(),
-            Word::random_linear_combine_expr(
-                (*EMPTY_HASH_LE).map(|byte| byte.expr()),
-                cb.power_of_randomness(),
-            ),
-        );
-
         // Sum up and verify gas cost.
-        let gas_cost = select::expr(
+        // Only CALL opcode could invoke transfer to make empty account into non-empty.
+        let gas_cost = call_gadget.gas_cost_expr(
             is_warm_prev.expr(),
-            GasCost::WARM_ACCESS.expr(),
-            GasCost::COLD_ACCOUNT_ACCESS.expr(),
-        ) + has_value.clone()
-            * (GasCost::CALL_WITH_VALUE.expr()
-                // Only CALL opcode could invoke transfer to make empty account into non-empty.
-                + is_call.expr() * (1.expr() - callee_exists.expr()) * GasCost::NEW_ACCOUNT.expr())
-            + memory_expansion.gas_cost();
-
+            // has_value.clone(),
```

Comment:
remove this line .

Reply:
fixed in [7596a17](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1023/commits/7596a17e82b7f5d61b7611aca985419b4c2bb5a9)